addons/mail/static/src/core/web/mail_core_web_service.js

```diff
@@ -41,7 +41,7 @@ export class MailCoreWeb {
             if (message.thread && notifId > message.thread.message_needaction_counter_bus_id) {
                 message.thread.message_needaction_counter++;
             }
-            message.thread?.notifyMessageToUser(message);
+            this.store.env.services["mail.out_of_focus"].notify(message, this);
```

Comment:
nit-pick: I think the precious one was better and more readable and had a clearer API.

Reply:
`notifyMessageToUser` targets a thread, this means the message has to be linked to a thread.
Inbox messages are not necessarily linked to a thread, e.g. `You have been assigned to` messages.
We should also put them as contribution to the out-of-focus counter, as they are inbox messages.

The `this` is wrong as 2nd parameter, I'll fix it!